### PR TITLE
Support analysis JSON format changes in the Atlas 2.8

### PR DESCRIPTION
**Pull request by acumarav**
Support analysis JSON format changes in the Atlas 2.8

### CODE REVIEW

File: src/main/java/com/odysseusinc/arachne/datanode/dto/serialize/NameDeserializer.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonDeserializer;
+import com.fasterxml.jackson.databind.JsonNode;
+import org.apache.commons.lang3.StringUtils;
+
+import java.io.IOException;
+import java.util.Objects;
+
+public class NameDeserializer extends JsonDeserializer<String> {
```
[the remark this answers is not in the document]
Comment (acumarav) [reply src/main/java/com/odysseusinc/arachne/datanode/dto/serialize/NameDeserializer.java:34]: Done

File: src/main/java/com/odysseusinc/arachne/datanode/dto/serialize/NameDeserializer.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+    private static final String NAME_FIELD = "name";
+
+    @Override
+    public String deserialize(JsonParser jsonParser, DeserializationContext deserializationContext) throws IOException {
```
Comment (YaroslavTir) [inline src/main/java/com/odysseusinc/arachne/datanode/dto/serialize/NameDeserializer.java:39]: Maybe it is better to slightly  change the code this way:

```public class NameDeserializer extends JsonDeserializer<String> {

    private static final String NAME_FIELD = "name";

    @Override
    public String deserialize(JsonParser jsonParser, DeserializationContext deserializationContext) throws IOException {

        if (Objects.isNull(jsonParser) || StringUtils.isBlank(jsonParser.getText())) {
            return null;
        }

        JsonNode node = jsonParser.getCodec().readTree(jsonParser);

        if (isAtlasVersionLessOrEquals27(node)) {
            return node.asText();
        }
        if (isAtlasVersionGraterOrEquals28(node)) {
            return node.get(NAME_FIELD).asText();
        }
        return null;
    }

    private boolean isAtlasVersionGraterOrEquals28(JsonNode node) {

        if (node.has(NAME_FIELD)) {
            return true;
        }
        return false;
    }

    private boolean isAtlasVersionLessOrEquals27(JsonNode node) {

        if (node.isValueNode()) {
            return true;
        }
        return false;
    }
}
```
and add JDoc, that in the future, we can replace NameDeserializer class with a  class 
```
class User {
   String  id;
   String name;
   String login;
   
}
```

File: src/main/java/com/odysseusinc/arachne/datanode/dto/serialize/NameDeserializer.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (YaroslavTir) [reply src/main/java/com/odysseusinc/arachne/datanode/dto/serialize/NameDeserializer.java:34]: I would rename `NameDeserializer` to `UserNameDeserializer`

File: src/main/java/com/odysseusinc/arachne/datanode/dto/serialize/NameDeserializer.java (hunk 2 above)
Comment (acumarav) [reply src/main/java/com/odysseusinc/arachne/datanode/dto/serialize/NameDeserializer.java:39]: I do not think that is it a good idea to try to backtrace atlas version based on provided json. The parser supports string and object an parse them accordingly. All those nested calls just complicate reading.

**Review by YaroslavTir — APPROVED** [review]

**Pull request merged**